src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -1293,6 +1299,161 @@ private function registerPropertyAccessConfiguration(array $config, ContainerBui
         ;
     }
 
+    private function registerAmqpConfiguration(array $config, ContainerBuilder $container)
+    {
+        $defaultConnectionName = null;
+        if (isset($config['default_connection'])) {
+            $defaultConnectionName = $config['default_connection'];
+        }
+
+        $match = false;
+        foreach ($config['connections'] as $name => $connection) {
+            $container
+                ->register("amqp.broker.$name", Broker::class)
+                ->addArgument($connection['url'])
+                ->addArgument($connection['queues'])
+                ->addArgument($connection['exchanges'])
+                ->setPublic(false)
+            ;
+            if (!$defaultConnectionName) {
+                $defaultConnectionName = $name;
+            }
+            if ($defaultConnectionName === $name) {
+                $match = true;
+            }
+        }
+
+        if (!$match) {
+            throw new \InvalidArgumentException("The default_connection \"$defaultConnectionName\" does not exist.");
+        }
+
+        $container->setAlias('amqp.broker', "amqp.broker.$defaultConnectionName");
+    }
+
+    private function registerWorkerConfiguration(array $config, ContainerBuilder $container)
+    {
+        $fetchers = array();
+
+        foreach ($config['fetchers']['amqps'] as $name => $fetcher) {
+            if (isset($fetchers[$name])) {
+                throw new \InvalidArgumentException("A fetcher named \"$name\" already exist.");
```

Comment:
> Please use `sprintf()` instead of variable interpolation in error messages.

Variable interpolation is optimized in PHP7, maybe starting using this kind of practice is nice?